### DISCUSSION TITLE
Any documents about how to make tauri 2.0 support Win 7?

### DISCUSSION THREAD

**Discussion by crackleeeessyp** [tauri-apps/tauri#12016] [Q&A] [answered]
Are there any documents that showing how to make tauri 2.0 support windows7? I tried latest version, it shows error:

![image](https://github.com/user-attachments/assets/50637cba-6046-4af5-b7f5-9edf19d22ba1)

Any suggestions such as rust version, dependencies version, or configurations I needs to adjust to make it work?

**Comment by FabianLars** [comment, marked as answer]
> rust version

For rust use 1.77.2 - it's the last one with full win 7 support. After that you'll have to use the special windows 7 rust target (`<arch>-win7-windows-msvc`) but it's not Tier 1 anymore.
Edit: This ^ should fix the error in your screenshot.

> dependencies version

Tauri v1 and v2 and all the plugins should still support win 7 as they are, including "inner" dependencies.

> or configurations

https://v2.tauri.app/distribute/windows-installer/#supporting-windows-7 - On top of that, don't use `transparent` windows. That's the only thing that comes to mind right now.

**Reply by A25468451** [reply]
hi. I tried to compile tauri to win7。However, the following exception occurs。
`error: package `cargo-platform v0.1.9` cannot be built because it requires rustc 1.78 or newer, while the currently active rustc version is 1.77.2
`

Use the cargo tree command

discover  tauri-utils  use  cargo-platform v0.1.9

`

├── tauri-utils v2.1.0
│   │   │   │   ├── brotli v7.0.0 (*)
│   │   │   │   ├── cargo_metadata v0.18.1
│   │   │   │   │   ├── camino v1.1.9
│   │   │   │   │   │   └── serde v1.0.216 (*)
│   │   │   │   │   ├── cargo-platform v0.1.9
│   │   │   │   │   │   └── serde v1.0.216 (*)
│   │   │   │   │   ├── semver v1.0.24 (*)
│   │   │   │   │   ├── serde v1.0.216 (*)
│   │   │   │   │   ├── serde_json v1.0.134 (*)
│   │   │   │   │   └── thiserror v1.0.69 (*)

`

this my Cargo.toml

` 
[dependencies]
tauri = { version = "2.1.1", features = [ "tray-icon", "image-png"] }
window-shadows = "0.2.2"
tokio = { version = "1.41.1", features = ["full"] }
tauri-plugin-os = "2"
tauri-plugin-notification = { version = "2.0.0", features = [ "windows7-compat" ] }
tauri-plugin-fs =  { version = "2.0.0", features = ["watch"] }
tauri-plugin-dialog = "2"
tauri-plugin-process = "2"
tauri-plugin-log = "2.0.0"
tauri-plugin-autostart = "2.0.1"
tauri-plugin-single-instance = "2.0.1"
tauri-plugin-updater = "2"
tauri-plugin-window-state = "2"
tauri-plugin-shell = "2"

`



What am I supposed to do now?

**Reply by FabianLars** [reply]
If that happens you'll have to use cargo update to downgrade these packages in your lockfile. For this package it's this for example
```
cargo update -p cargo-platform --precise 0.1.8
```

I assume there will be more, so to do the rest yourself, simply look up the versions here https://crates.io/crates/cargo-platform/versions (replacing cargo-platform with the crate you need). If the crate didn't specify `rust-version` you often can check out the changelog in their github repo. If that's also not helpful you'll have to do trial and error.

**Reply by A25468451** [reply]
That's OK now. Thank you for your reply
